client/core/core.go

```diff
@@ -1513,36 +1552,38 @@ func (c *Core) notifyFee(dc *dexConnection, coinID []byte) error {
 		return fmt.Errorf("failed to create notifyfee request: %v", err)
 	}
 
+	// Make the 'notifyfee' request and wait for the response. The server waits
+	// an unspecified amount of time to discover the transaction, so time out
+	// after this DEX's configured broadcast timeout plus a healthy buffer for
+	// communication and server processing latencies. There is no reason to give
+	// up too soon.
+	timeout := time.Millisecond*time.Duration(dc.cfg.BroadcastTimeout) + 10*time.Second
 	errChan := make(chan error, 1)
-	err = dc.Request(msg, func(resp *msgjson.Message) {
+	err = dc.RequestWithTimeout(msg, func(resp *msgjson.Message) {
 		ack := new(msgjson.Acknowledgement)
-		err := resp.UnmarshalResult(ack)
-		if err != nil {
+		// Do NOT capture err in this closure.
+		if err := resp.UnmarshalResult(ack); err != nil {
 			errChan <- fmt.Errorf("notify fee result error: %v", err)
 			return
 		}
-		// If there was a serialization error, validation is skipped. A warning
-		// message was already logged.
-		if len(reqB) > 0 {
```

Comment:
I'm also confused by why/how `reqB` could ever have been empty, and if it were why that would be OK.